pkg/ec2provider/ec2provider.go

```diff
@@ -277,3 +285,41 @@ func (p *ec2ProviderImpl) getPrivateDnsAndPublishToCache(instanceIdList []string
 		p.unsetRequestInFlightForInstanceId(id)
 	}
 }
+
+func applySTSRequestHeaders(stsClient *sts.STS, roleARN, sourceARN string) *sts.STS {
+	logrus.Infof("Using AWS assumed role %v", roleARN)
+	sourceAcct, err := getSourceAccount(roleARN)
```

Comment:
Shouldn't the account ID be pulled from the sourceARN and not the role ARN?

Reply:
They _should_ be the same, but in the event of a config mix up, which one would we want to set?

Reply:
We will use the roleARN (ec2DescribeInstancesRoleARN) because this is the role we are going to assume. We did same in CCM https://github.com/kubernetes/cloud-provider-aws/pull/649

Reply:
Discussed offline, we are gonna parse source account id from sourceARN to avoid confusion deputy issue in the misconfiguration scenario. Also panic the authenticator when the sourceARN is ill-formatted